### DISCUSSION TITLE
Flat Config and git submodules monorepo

### DISCUSSION THREAD

**Discussion by nashspence** [eslint/eslint#17451] [Q&A]
I use a setup that I like that involves creating a monorepo with workspaces that are also git submodules. This allows these submodules to function both as part of the monorepo and standalone.

I like to use the same eslint-config across all of these submodules. In the past, I have done this by creating an eslint-config package and setting the eslint->extends to point to that package in the package.json of all of my other packages. This worked great. VSCode tooling and the eslint CLI could find the right config anywhere.

Is there any analogous way to achieve this with the new system? I have seen somewhat similar discussions, but none dealing particularly with a monorepo structured in this way. The key here is that it is not an option to use one config above everything in the monorepo package, but it is very undesirable to create or copy a separate config into each of the submodules.

**Comment by Rec0iL99** [comment]
Hi @nashspence, I suppose you'll need to continue having different configs (`eslint.config.js`) in each submodule, extending the root config for now. Once the new config system becomes widely adopted by the community, we'll have more examples and discussions to explore regarding git submodules.

**Reply by nashspence** [reply]
That's what I was thinking in the end. It appears I'm on the wrong side of progress this time around. Thanks for the reply. Yah, it would be my hope that somehow it would be supported again or something similar, though I get it if no one else likes doing it this way then it might not be worth it.
